src/sql/magic.py

```diff
@@ -505,6 +533,19 @@ def interactive_execute_wrapper(**kwargs):
                     + " instead for the save argument.",
                     FutureWarning,
                 )
+            if self.persist_snippets:
```

Comment:
this magic.py file is getting busy, please move this logic to store.py, it's better to put it there. the only thing that should be here are the conditionals that determine whether to store the snippets or not based on the config.persist_snippets flag

Reply:
Sure! got it

Reply:
moved the logic to store.py